lib/matplotlib/axes/_axes.py

```diff
@@ -7548,8 +7551,10 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
         xmin, xmax = xextent
         freqs += Fc
         extent = xmin, xmax, freqs[0], freqs[-1]
+
+        kwargs.pop('origin', None)
```

Comment:
```suggestion
        if 'origin' in kwargs:
            raise TypeError("specgram() got an unexpected keyword argument "
                            "'origin'")
```
to make the function fail like with any unsupported keyword argument and not silently ignore it.

Technically, we should even just raise a deprecation warning, but since the behavior was broken, there's little point in keeping the broken functionality with a warning.